src/main/java/org/reaktivity/nukleus/sse/internal/stream/ServerStreamFactory.java

```diff
@@ -773,6 +793,46 @@ private void handleReset(
             final long traceId = reset.trace();
             doReset(applicationReplyThrottle, applicationRouteId, applicationReplyId, traceId);
         }
+
+        // TODO - SseEventFW
+        //                  .type = "challenge"
+        //                  .data = "{}"
+        // Received signalEx CHALLENGE from OAuth
+        private void handleSignal(
+            SignalFW signal)
+        {
+            final HttpSignalExFW httpSignalEx = httpSignalExRO.tryWrap(signal.buffer(),
+                                                                       signal.extension().offset(),
+                                                                       signal.extension().limit());
+            if (httpSignalEx != null && signal.extension().sizeof() == httpSignalExRO.limit())
+            {
+                final ListFW<HttpHeaderFW> httpHeaders = httpSignalEx.headers();
+                final Map<String, String> collectedHeaders = new HashMap<>();
+                httpHeaders.forEach(header -> collectedHeaders.put(header.name().asString(), header.value().asString()));
+
+                // TODO: need ":method" -> "method"
+                final JsonObject jsonHeaders = new JsonObject();
+                collectedHeaders.forEach(jsonHeaders::addProperty);
+                final String headersPayload = gson.toJson(jsonHeaders);
+
+                final JsonObject payload = new JsonObject();
+                payload.addProperty("headers", headersPayload);
+                final String jsonPayload = gson.toJson(payload);
+
+                final StringFW challengeType = stringRW.wrap(writeBuffer, 0, writeBuffer.capacity())
+                        .set("challenge", UTF_8)
+                        .build();
+                final StringFW data = stringRW.wrap(writeBuffer, 0, writeBuffer.capacity())
+                        .set(jsonPayload, UTF_8)
+                        .build();
+                final OctetsFW challengeData = octetsRW.set(data.buffer(), 0, data.limit()).build();
+
+                final SseEventFW sseEvent = sseEventRW.wrap(writeBuffer, DataFW.FIELD_OFFSET_PAYLOAD, writeBuffer.capacity())
+                        .type(challengeType.value())
+                        .data(challengeData)
```

Comment:
Change signature of `SseEventFW.data(...)` method to receive `Flyweight` instead of `OctetsFW` and pass `data` as `StringFW` directly.